docs/src/main/asciidoc/se/guides/logging.adoc

```diff
@@ -0,0 +1,239 @@
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jul-to-slf4j</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>${version.lib.logback}</version>
+        </dependency>
+----
+
+The above snippet uses logback as the slf4j logging provider.
+
+This achieves four things:
+
+1. Makes the slf4j API available to your application.
+2. Bridges all of Helidon's Java Util Logging to slf4j
+3. Enables the use of MDC with slf4j
+4. Ensures GraalVM native-image support for slf4j
+
+For more information see link:{helidon-github-examples-url}/logging/slf4j[Helidon slf4j example]
+
+== Mapped Diagnostic Context (MDC) support
+
+The Mapped Diagnostic Context provides a mechanism for Helidon and application code to store
+values in a context that can then be included in log output. These values could
+be a tracing ID, the current user, or a similar contextual value, and they are
+propagated by Helidon across threads.
+
+Some logging frameworks, such as slf4j and log4j, have support for MDC. But
+Java Util Logging does not. Therefore, Helidon provides an API, `HelidonMdc`,
+that you can use to store contextual values. `HelidonMdc` not only works with JUL
+but also works with the other supported frameworks (slf4j or log4j).
+
+To use Helidon MDC, you need to do three things:
+
+1. Add a dependency on one of the supported logging frameworks. See previous sections.
+2. Store values in the MDC in your code.
+3. Configure logging to display the values in its output.
+
+=== MDC values provided by Helidon
+If you include tracing in your project, Helidon automatically provides the MDC key `trace_id`. Logging inserts the trace ID of the current span (if there is one) or the string "none" if no span is active.
+
+=== Storing values in MDC
+
+To set your own values in the MDC you can use `io.helidon.logging.common.HelidonMdc` class:
+
+[source,java]
+----
+HelidonMdc.set("name", "Joe"); // <1>
+HelidonMdc.set("otherName", () -> retrieveStringValue()); // <2>
+----
+<1> Stores an attribute named "name" with the fixed value "Joe" in the MDC.
+<2> Stores an attribute named "otherName" with the given `Supplier<String>`.
+
+The `set(<String, String>`) method associates the name with a fixed value. The `set(<String>, <Supplier<String>>) form associates the name with a lambda or method reference. This second form might be useful if your code (or Helidon) maintains the value you want to log in some context that your code can access and if that value changes as the server runs. In both cases, when your logging format refers to the name logging extracts the corresponding fixed value or invoked the supplier to obtain the value, using the result in log output.
```

Comment:
change "invoked the supplier" to "invokes the supplier"

Reply:
Also note the missing backtick on line 183